FlowCrypt/Controllers/Setup/SetupInitialViewController.swift

```diff
@@ -158,30 +158,32 @@ extension SetupInitialViewController {
         Task {
             do {
                 let idToken = try await IdTokenUtils.getIdToken(userEmail: appContext.user.email)
-                let result = try await emailKeyManagerApi.getPrivateKeys(idToken: idToken)
-                switch result {
-                case .success(keys: let keys):
-                    proceedToSetupWithEKMKeys(keys: keys)
-                case .noKeys:
-                    showRetryAlert(
-                        message: "organisational_rules_ekm_empty_private_keys_error".localized,
-                        onRetry: { [weak self] in
-                            self?.state = .fetchingKeysFromEKM
-                        },
-                        onOk: { [weak self] in
-                            self?.signOut()
-                        }
-                    )
-                case .keysAreNotDecrypted:
-                    showAlert(
-                        message: "organisational_rules_ekm_keys_are_not_decrypted_error".localized,
-                        onOk: { [weak self] in
-                            self?.signOut()
-                        }
-                    )
-                }
+                let keys = try await emailKeyManagerApi.getPrivateKeys(idToken: idToken)
+                proceedToSetupWithEKMKeys(keys: keys)
             } catch {
-                if case .noPrivateKeysUrlString = error as? EmailKeyManagerApiError {
+                if let ekmError = error as? EmailKeyManagerApiError {
+                    let errorMessage = ekmError.errorMessage
+                    switch ekmError {
+                    case .noKeys:
+                        showRetryAlert(
+                            message: errorMessage,
+                            onRetry: { [weak self] in
+                                self?.state = .fetchingKeysFromEKM
+                            },
+                            onOk: { [weak self] in
+                                self?.signOut()
+                            }
+                        )
+                    case .keysAreNotDecrypted, .keysAreInvalid:
+                        showAlert(
+                            message: errorMessage,
+                            onOk: { [weak self] in
+                                self?.signOut()
+                            }
+                        )
+                    case .noPrivateKeysUrlString:
+                        break
+                    }
```

Comment:
All of the situations in this catch could be handled exactly the same way, with an `ok` and `retry` button as follows:

```swift
                        showRetryAlert(
                            message: errorMessage,
                            onRetry: { [weak self] in
                                self?.state = .fetchingKeysFromEKM
                            },
                            onOk: { [weak self] in
                                self?.signOut()
                            }
                        )
```

That means we probably wouldn't need `if let ekmError = error as? EmailKeyManagerApiError {` and we also wouldn't need the switch. It can all be handled the same.

Reply:
I'm unsure why this handler evolved to have three ways to handle the error, I think one way is enough, as long as it prints the appropriate error message.

Reply:
Yeah, that's how it was written, but there wasn't any reason for it that I know.

Reply:
Got it